bin/inx-chronicle/src/main.rs

```diff
@@ -128,12 +151,18 @@ impl HandleEvent<Report<InxListener>> for Launcher {
                 cx.shutdown();
             }
             Err(e) => match &e.error {
-                ActorError::Result(e) => match e.downcast_ref::<InxListenerError>().unwrap() {
+                ActorError::Result(e) => match e.downcast_ref::<<InxListener as Actor>::Error>().unwrap() {
                     InxListenerError::Inx(e) => match e {
-                        InxError::TransportFailed => {
-                            cx.spawn_actor_supervised(InxListener::new(config.inx.clone(), broker_addr.clone()))
-                                .await;
-                        }
+                        // TODO: This is stupid, but we can't use the ErrorKind enum so :shrug:
+                        InxError::TransportFailed(e) => match e.to_string().as_ref() {
```

Comment:
Could this be cleaned up by using `e.source()`?

Reply:
Not as far as I know. It returns a `dyn Error` which is even more useless.